EIPS/eip-5089.md

```diff
@@ -0,0 +1,340 @@
+
+MUST _NOT_ revert.
+
+```yaml
+- name: maturityRate
+  type: function
+  stateMutability: view
+
+  inputs: []
+
+  outputs:
+    - name: rate
+      type: uint256
+```
+
+#### exchangeRate
+
+The current exchangeRate of the underlying token <-> money market deposit post-maturity. Returns 0 before maturity.
+_Alberto: I wonder if we could make the ERC a bit more compact, and have only `exchangeRate` (renamed to `rate`). Before maturity, it returns the rate that will work at maturity, after maturity, it can return the same, or change._
+
+OPTIONAL - This method can be used to provide needed data for the calculation of interest post-maturity, but interfaces and other contracts MUST NOT expect this method to be present. _Alberto: Optionality in ERCs is better avoided. Not everyone will implement it, therefore integrators must assume it might not be there, and leads to a proliferation of `safe*` methods._
+
+MUST _NOT_ revert.
+
+```yaml
+- name: exchangeRate
+  type: function
+  stateMutability: view
+
+  inputs: []
+
+  outputs:
+    - name: rate
+      type: uint256
+```
+
+#### mint
+
+_Alberto: do we need a `mint` method? I reckon it might be nice for integrators to have a standard function to obtain the PTs, but so far we have been fine without one for ERC20s._
+
+Mints exactly `amount` of Principal Tokens to `to` by depositing that exact same `amount` of underlying tokens. In yield tokenization protocols, also mints `amount` of Yield Tokens in addition to Principal Tokens.
+
+Optional - This method can be used to improve the usability and integration difficulty of Principal Tokens, but interfaces and other contracts MUST NOT expect custodial methods (mint/redeem) to be present in the Principal Token contract.
+While custodial minting of Principal Tokens through the Principal Token contract is extremely useful for integrators, some protocols may find giving the Principal Token itself custody breaks their backwards compatability. 
+
+MUST emit the `Mint` event.
+
+MUST support ERC-20 `approve` / `transferFrom` on `asset` as a mint flow.
+MAY support an additional flow in which the underlying tokens are owned by the Principal Token contract before the `mint` execution, and are accounted for during `mint`.
+
+MUST revert if all of `amount` cannot be minted (due to deposit limit being reached, slippage, the user not approving enough underlying tokens to the PT contract, etc).
+
+Note that most implementations will require pre-approval of the Principal Token contract with the Principal Tokens `underlying` token.
+
+```yaml
+- name: mint
+  type: function
+  stateMutability: nonpayable
+
+  inputs:
+    - name: amount
+      type: uint256
+    - name: to
+      type: address
+
+  outputs:
+    - name: success
+      type: bool
+```
+
+#### redeem
+
+After maturity, burns exactly `amount` of Principal Tokens from `from` and sends the _same_ `amount` of underlying tokens to `to`.
+_Alberto: It sends underlying tokens according to `exchangeRate`._
```

Comment:
>  Optionality is evil.

I didn't want to force *every* protocol's implementation to distribute yield post maturity, would this mean that these protocols are just expected to set an `exchangeRate` of 1:1 at maturity?

Seems like an ok option, though it would mean compliance with the standard just for compliance's sake.

Anywho, assuming thats the direction the EIP goes, then yeah 100% on board with injecting the language everywhere.

> After maturity, burns exactly `amount` of Principal Tokens from `from` and sends the _same_ `amount` of underlying tokens to `to`.

-> 

After maturity, burns exactly `amount` of Principal Tokens from `from` and sends underlying tokens according of underlying tokens to the current exchangeRate.